src/main.zig

```diff
@@ -1664,3 +1670,111 @@ test "Test happy flow" {
         }
     }
 }
+
+// Code to test NIST Known Answer Tests (KAT), see PQCgenKAT.c.
+
+const sha2 = std.crypto.hash.sha2;
+
+test "NIST KAT test" {
+    inline for (modes) |mode| {
+        var seed: [48]u8 = undefined;
+        var i: usize = 0;
+        while (i < seed.len) : (i += 1) {
+            seed[i] = @intCast(u8, i);
+        }
+        var f = sha2.Sha256.init(.{});
+        var g = NistDRBG.new(&seed);
+        try std.fmt.format(f, "# {}\n\n", .{mode.name});
+        i = 0;
+        while (i < 100) : (i += 1) {
+            g.fill(&seed);
+            try std.fmt.format(f, "count = {}\n", .{i});
+            try std.fmt.format(f, "seed = {X}\n", .{seed});
+            var g2 = NistDRBG.new(&seed);
+
+            // This is not equivalent to g2.fill(kseed[:]).  As the reference
+            // implementation calls randombytes twice generating the keypair,
+            // we have to do that as well.
+            var kseed: [64]u8 = undefined;
+            var eseed: [32]u8 = undefined;
+            g2.fill(kseed[0..32]);
+            g2.fill(kseed[32..64]);
+            g2.fill(&eseed);
+            const kp = mode.keyFromSeed(&kseed);
+            var ct: [mode.ciphertextSize]u8 = undefined;
+            var ss: [sharedKeySize]u8 = undefined;
+            kp.pk.encapsDeterministically(eseed, &ct, &ss);
+            const ss2 = kp.sk.decaps(&ct);
+            testing.expectEqual(ss2, ss);
+            try std.fmt.format(f, "pk = {X}\n", .{kp.pk.pack()});
+            try std.fmt.format(f, "sk = {X}\n", .{kp.sk.pack()});
+            try std.fmt.format(f, "ct = {X}\n", .{ct});
+            try std.fmt.format(f, "ss = {X}\n\n", .{ss});
+        }
+
+        var out: [32]u8 = undefined;
+        f.final(&out);
+        std.io.getStdOut().writer().print("{X}\n", .{out});
+    }
+}
+
+const NistDRBG = struct {
+    key: [32]u8,
+    v: [16]u8,
+
+    fn incV(g: *NistDRBG) void {
+        var j: usize = 15;
+        while (j >= 0) : (j -= 1) {
+            if (g.v[j] == 255) {
```

Comment:
`v[x]` is a `u8` so you could just do `g.v[j] +%= 1`

Reply:
We still need the branch to decide whether to break. Perhaps the most elegant approach is to use an `u128`.

Reply:
Oh right. Didn't see the `break` :) I'd really need to take one.

Yep, we could just use a `u128`.